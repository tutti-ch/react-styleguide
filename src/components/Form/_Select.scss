@import "../../styles/Colors/_colors";
@import "../../styles/Sizes/_sizes";
@import "../../styles/Motion/_transitions";
@import "../../styles/Borders/_borders";
@import "../../styles/Shadows/_shadows";

.wrapper {
  position: relative;
  outline: none;

  .label {
    color: $greyscale-60;
  }

  &:not(.inline) {
    .label {
      transition: all $transition-m;
      left: $size-xs;
    }

    &.hasValue .label {
      font-size: $size-s;
      top: -$size-m;
      left: 0;
    }
  }

  .wrapperInner {
    border: $border-default;
    background: $greyscale-100;
    height: $size-xxxl;
    line-height: $size-xxxl;
    position: relative;
    text-align: left;
    padding: 0 $size-xs;
    cursor: pointer;

    &:focus {
      border-color: $greyscale-00;
      outline: none;
    }

    .options {
      background: $greyscale-100;
      position: absolute;
      margin-top: -2px;
      margin-bottom: 4rem; // When the options hit the bottom of page, this gives a bit of space
      top: 100%;
      right: -1px;
      left: -1px;
      opacity: 0;
      max-height: 0;
      visibility: hidden;
      overflow: auto;
      @include mediumShadow();
      border: $border-default;
      user-select: none;

      // Keep the style separated for the placeholder option
      .option {
        border-bottom: 1px solid $greyscale-95;
        padding: 0 $size-xs;

        &:hover,
        &.highlighted {
          background-color: $blue-95;
        }

        &.selected {
          font-weight: bold;
        }
      }
    }

    .ph {
      &:after {
        content: "";
        border-color: $greyscale-80 transparent transparent;
        border-style: solid;
        border-width: 0.4rem 0.4rem 0.2rem;
        display: inline-block;
        height: 0;
        width: 0;
        transition: all $transition-m;
        right: $size-xs;
        margin-top: 0.2rem;
      }

      .option {
        padding-right: $size-xs;
      }
    }

    .option {
      display: flex;
      align-items: center;
    }
  }

  .icon {
    margin-right: $size-xs;
    width: $size-xl;
    height: $size-xl;
  }

  // When the select box is open or is hover make the border black
  &.is-open .options,
  &.is-open .wrapperInner,
  &:not(.disabled):not(.inline):hover .wrapperInner {
    border-color: $greyscale-00;
  }

  &.is-open {
    .options {
      max-height: 320px;
      opacity: 1;
      visibility: visible;
    }

    &:not(.inline) {
      border-top-color: $greyscale-80;
    }
  }

  &:not(.disabled):hover .ph:after,
  &.is-open .ph:after {
    border-color: $greyscale-00 transparent transparent;
  }

  &.disabled {
    .wrapperInner {
      cursor: default;
      background: $greyscale-95;
      color: $greyscale-60;
    }

    .icon {
      opacity: 0.6;
    }
  }

  // Styles only for inline
  &.inline {
    &,
    .label,
    .wrapperInner {
      display: inline-block;
    }

    .wrapperInner {
      border: 1px solid transparent;
      display: inline-block;
      background-color: transparent;
      height: auto;
      line-height: inherit;

      .options .option {
        line-height: $size-xxxl;
        height: $size-xxxl;
      }
    }

    .ph:after {
      right: 0;
    }

    .options {
      left: auto;
      margin-top: $size-m;
      min-width: 250px;
      max-height: 320px;
    }
  }
}

// Style for inputs props.size='small'
.small {
  padding-top: 0;
  padding-bottom: $size-xl;

  .errorMessage {
    top: $size-xxxl;
    line-height: $size-m;
  }
  .label {
    top: 80 / 2;
  }

  .input:focus ~ .label,
  .label.filled {
    font-size: $size-s;
    top: - $size-s;
    left: 0;
  }
}
