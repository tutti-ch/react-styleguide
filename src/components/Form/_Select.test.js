/* global describe, test, expect, jest */
import React from "react";
import { mount } from "enzyme";
import { Select } from "./_Select";

describe("(Component) Select", () => {
  const getOptions = () => [
    { value: "React.js", text: "React.js" },
    { value: "Vue.js", text: "Vue.js" },
    { value: "Angular.js", text: "Angular.js" },
    { value: "jQuery", text: "jQuery" },
    { value: "Knockout", text: "Knockout" },
    { value: "Backbone", text: "Backbone" },
    { value: "Ember", text: "Ember" },
    { value: "Bootstrap", text: "Bootstrap" },
    { value: "Other", text: "Other" }
  ];

  // test("[componentDidUpdate] should update the state when the component receives new props", () => {
  //   const options = getOptions();
  //   const comp = mount(<Select options={options} selected="jQuery" />);
  //   expect(comp.state("selected")).toEqual(["jQuery"]);

  //   const spy = jest.spyOn(comp.instance(), "sortOptions");
  //   comp.setProps({ selected: ["jQuery", "Ember"], multiple: true });

  //   expect(comp.state("selected")).toEqual(["jQuery", "Ember"]);
  //   expect(spy).not.toHaveBeenCalled();

  //   comp.setProps({
  //     options: [{ value: "NewF.js", text: "New Framework" }],
  //     sort: true
  //   });
  //   expect(spy).toHaveBeenCalled();

  //   const setStateSpy = jest.spyOn(comp.instance(), "setState");
  //   comp.setProps({});
  //   expect(setStateSpy).not.toHaveBeenCalled();

  //   comp.setProps({ options, sort: false, selected: "React.js" });
  //   expect(setStateSpy).toHaveBeenCalledWith({
  //     highlighted: -1,
  //     selected: ["React.js"],
  //     options: options
  //   });
  // });

  test("[componentDidUpdate] should handle highlighted state", () => {
    const options = getOptions();
    const comp = mount(<Select options={options} selected="jQuery" />);

    expect(comp.state("selected")).toEqual(["jQuery"]);
    expect(comp.state("highlighted")).toBe(3);
    const newOptions = getOptions();

    newOptions[3].value = "nojQuery";
    comp.setProps({
      options: newOptions,
      selected: undefined
    });
    expect(comp.state("highlighted")).toBe(-1);
    expect(comp.state("selected")).toEqual([]);
  });

  test("[componentDidUpdate] should disable highlighted state when multiline", () => {
    const options = getOptions();
    const comp = mount(<Select options={options} selected="jQuery" />);

    comp.setProps({
      selected: ["Ember"],
      multiple: true
    });

    expect(comp.state("selected")).toEqual(["Ember"]);
    expect(comp.state("highlighted")).toBe(-1); // does not support multiple
  });

  test("[handleArrowDown/handleArrowUp] should increase/decrease the highlight index", () => {
    const options = getOptions();
    const comp = mount(<Select options={options} sort />);
    const inst = comp.instance();
    expect(comp.state("highlighted")).toBe(-1);
    inst.handleArrowDown();
    expect(comp.state("highlighted")).toBe(0);
    inst.handleArrowDown();
    expect(comp.state("highlighted")).toBe(1);
    inst.handleArrowUp();
    expect(comp.state("highlighted")).toBe(0);
    inst.handleArrowUp();
    expect(comp.state("highlighted")).toBe(0);
    comp.setState({ highlighted: options.length - 1 });
    inst.handleArrowDown();
    inst.handleArrowDown();
    expect(comp.state("highlighted")).toBe(options.length - 1);
  });

  test("[handleKeyDown] should handle several keys", () => {
    const options = getOptions();
    const comp = mount(
      <Select options={options} selected={["jQuery", "Ember"]} sort />
    );
    const inst = comp.instance();

    comp.setState({ highlighted: 1 });

    const event = {
      key: "ArrowDown",
      preventDefault: jest.fn()
    };

    inst.handleArrowUp = jest.fn();
    inst.handleArrowDown = jest.fn();
    inst.select = jest.fn();
    const spy = jest.spyOn(inst, "open");

    // The first stroke should open the options
    inst.keyDown(event);
    expect(event.preventDefault).toHaveBeenCalled();
    expect(spy).toHaveBeenCalled();
    expect(inst.handleArrowDown).not.toHaveBeenCalled();

    inst.keyDown(event);
    expect(inst.handleArrowDown).toHaveBeenCalled();

    event.key = "Up";
    inst.keyDown(event);
    expect(inst.handleArrowUp).toHaveBeenCalled();

    event.key = "Enter";
    inst.keyDown(event);
    expect(inst.select).toHaveBeenCalledWith(options[1], event);
    expect(spy).toHaveBeenCalledTimes(1);

    event.key = "Esc";
    inst.keyDown(event);
    expect(comp.state("isOpen")).toBe(false);
  });

  test("[open] should set the state", () => {
    const options = getOptions();
    const comp = mount(
      <Select options={options} selected={["jQuery", "Ember"]} disabled />
    );
    const inst = comp.instance();

    inst.open();
    expect(comp.state("isOpen")).toBe(false);
    comp.setProps({ disabled: false });
    inst.open();
    expect(comp.state("isOpen")).toBe(true);
  });

  test("[close] should set the state", () => {
    const options = getOptions();
    const comp = mount(
      <Select options={options} selected={["jQuery", "Ember"]} />
    );
    const inst = comp.instance();
    const spy = jest.spyOn(inst, "setState");

    inst.close();
    expect(comp.state("isOpen")).toBe(false);
    expect(spy).not.toHaveBeenCalled();
    inst.open();
    inst.close();
    expect(comp.state("isOpen")).toBe(false);
    expect(spy).toHaveBeenCalled();
  });

  test("[toggle] should set the state", () => {
    const options = getOptions();
    const comp = mount(
      <Select options={options} selected={["jQuery", "Ember"]} />
    );
    const inst = comp.instance();

    expect(comp.state("isOpen")).toBe(false);
    inst.toggle();
    expect(comp.state("isOpen")).toBe(true);
    inst.toggle();
    expect(comp.state("isOpen")).toBe(false);
  });

  test("[select] should select the given value and set the highlighted item", () => {
    const options = getOptions();
    const comp = mount(<Select options={options} selected="Ember" />);
    const inst = comp.instance();

    expect(inst.props.options).toEqual(options);
    expect(comp.state("selected")).toEqual(["Ember"]);
    expect(inst.findIndexByValue("Ember")).toBe(6);
    expect(comp.state("highlighted")).toBe(6);
    inst.select({ value: "jQuery" });
    expect(comp.state("selected")).toEqual(["jQuery"]);
    expect(comp.state("highlighted")).toBe(3);
  });

  test("[select] should select the given value and set the highlighted item - multiple", () => {
    const options = getOptions();
    const onChange = jest.fn();
    const comp = mount(
      <Select
        options={options}
        onChange={onChange}
        selected={["Ember"]}
        name="favs"
        multiple
      />
    );
    const inst = comp.instance();
    const event = { stopPropagation: jest.fn() };

    expect(inst.props.options).toEqual(options);
    expect(comp.state("selected")).toEqual(["Ember"]);
    expect(inst.findIndexByValue("Ember")).toBe(6);
    expect(comp.state("highlighted")).toBe(-1); // Multiples are not highlighted
    inst.select({ value: "jQuery" }, event);
    expect(event.stopPropagation).toHaveBeenCalled();
    expect(comp.state("selected")).toEqual(["Ember", "jQuery"]);
    expect(comp.state("highlighted")).toBe(-1);
    expect(onChange).toHaveBeenCalledWith(["Ember", "jQuery"], {
      name: "favs",
      initialValue: ["Ember"],
      formValue: ["Ember", "jQuery"]
    });
    inst.select({ value: "jQuery" }, event);
    expect(comp.state("selected")).toEqual(["Ember"]);
    expect(comp.state("highlighted")).toBe(-1);
    expect(onChange).toHaveBeenCalledWith(["Ember"], {
      name: "favs",
      initialValue: ["Ember"],
      formValue: ["Ember"]
    });
  });

  test("[handleOnFocus/handleOnBlur] should add/remove the listener", () => {
    const comp = mount(<Select options={[]} />);
    const inst = comp.instance();

    const spyAdd = jest.spyOn(document, "addEventListener");
    const spyRem = jest.spyOn(document, "removeEventListener");

    inst.handleOnFocus();
    expect(spyAdd).toHaveBeenCalledWith("keydown", inst.keyDown);
    expect(comp.state().isFocused).toBe(true);

    inst.handleOnBlur();
    expect(spyRem).toHaveBeenCalledWith("keydown", inst.keyDown);
    expect(comp.state().isFocused).toBe(false);
  });

  test("[componentDidUpdate] should remove the listener if not focused", () => {
    const comp = mount(<Select options={[]} />);

    const spyRem = jest.spyOn(document, "removeEventListener");
    spyRem.mockClear();

    // if focused, we should not remove the event listener
    comp.setState({ isFocused: true });
    expect(spyRem).toHaveBeenCalledTimes(0);

    comp.setState({ isFocused: false });
    expect(spyRem).toHaveBeenCalledTimes(1);
  });

  test("[componentWillUnmount] should remove the listener", () => {
    const comp = mount(<Select options={[]} />);
    const inst = comp.instance();

    const spyRem = jest.spyOn(document, "removeEventListener");
    spyRem.mockClear();

    // causing a component update
    expect(spyRem).toHaveBeenCalledTimes(0);
    inst.componentWillUnmount();
    expect(spyRem).toHaveBeenCalledTimes(1);
  });

  test("[getSelectedOptions] should return the selected options or the placeholder if given", () => {
    const options = getOptions();
    const comp = mount(
      <Select options={options} selected={["React.js", "jQuery"]} multiple />
    );
    const inst = comp.instance();

    expect(inst.getSelectedOptions()).toEqual([options[0], options[3]]);
    comp.setProps({ placeholder: "my-placeholder", selected: [] });
    expect(inst.getSelectedOptions()).toEqual([
      { text: "my-placeholder", value: null }
    ]);

    expect(comp).toMatchSnapshot();
  });

  test("[unselect] should unselect the given value", () => {
    const onChange = jest.fn();
    const options = getOptions();
    const comp = mount(
      <Select
        options={options}
        selected={["React.js", "jQuery"]}
        multiple
        onChange={onChange}
      />
    );
    const inst = comp.instance();

    inst.unselect({ value: "React.js" });
    expect(comp.state("selected")).toEqual(["jQuery"]);
    inst.unselect({ value: "Ember" });
    expect(comp.state("selected")).toEqual(["jQuery"]);
    expect(onChange).toHaveBeenCalledWith(["jQuery"], {
      initialValue: ["jQuery"],
      name: undefined
    });
  });

  test("[resetSelected] should reset selected state", () => {
    const onChange = jest.fn();
    const options = getOptions();
    const comp = mount(
      <Select
        options={options}
        selected={["React.js", "jQuery"]}
        multiple
        onChange={onChange}
      />
    );
    const inst = comp.instance();

    inst.resetSelected();
    expect(comp.state("selected")).toEqual([]);
    expect(onChange).toHaveBeenCalledWith(null, {
      name: undefined,
      initialValue: ["React.js", "jQuery"]
    });
  });

  test("[snapshot] should match", () => {
    const options = getOptions();
    const comp = mount(<Select options={options} selected={["jQuery"]} />);
    expect(comp).toMatchSnapshot();
  });
});
