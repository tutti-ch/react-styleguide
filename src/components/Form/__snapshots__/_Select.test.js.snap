// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Select [getSelectedOptions] should return the selected options or the placeholder if given 1`] = `
<Select
  multiple={true}
  options={
    Array [
      Object {
        "text": "React.js",
        "value": "React.js",
      },
      Object {
        "text": "Vue.js",
        "value": "Vue.js",
      },
      Object {
        "text": "Angular.js",
        "value": "Angular.js",
      },
      Object {
        "text": "jQuery",
        "value": "jQuery",
      },
      Object {
        "text": "Knockout",
        "value": "Knockout",
      },
      Object {
        "text": "Backbone",
        "value": "Backbone",
      },
      Object {
        "text": "Ember",
        "value": "Ember",
      },
      Object {
        "text": "Bootstrap",
        "value": "Bootstrap",
      },
      Object {
        "text": "Other",
        "value": "Other",
      },
    ]
  }
  placeholder="my-placeholder"
  required={false}
  selected={Array []}
  size="medium"
>
  <OutsideClickable
    onOutsideClick={[Function]}
  >
    <div
      className="select multiple"
      key=".0"
      onClick={[Function]}
    >
      <div
        className="selectInner"
        onBlur={[Function]}
        onFocus={[Function]}
        tabIndex={0}
      >
        <div
          className="ph"
        >
          <Option
            closeIcon={true}
            key="React.js"
            onClose={[Function]}
            selected={false}
            text="React.js"
            value="React.js"
          >
            <div
              className="option"
              key="React.js"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                React.js
              </span>
              <Close
                className="icoClose"
                onClick={[Function]}
              >
                <Sprite
                  className="icoClose"
                  id="close"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <svg
                    className="svg-sprite icoClose"
                    onClick={[Function]}
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <use
                      xlinkHref="#close"
                    />
                  </svg>
                </Sprite>
              </Close>
            </div>
          </Option>
          <Option
            closeIcon={true}
            key="jQuery"
            onClose={[Function]}
            selected={false}
            text="jQuery"
            value="jQuery"
          >
            <div
              className="option"
              key="jQuery"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                jQuery
              </span>
              <Close
                className="icoClose"
                onClick={[Function]}
              >
                <Sprite
                  className="icoClose"
                  id="close"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <svg
                    className="svg-sprite icoClose"
                    onClick={[Function]}
                    role="img"
                    viewBox="0 0 24 24"
                  >
                    <use
                      xlinkHref="#close"
                    />
                  </svg>
                </Sprite>
              </Close>
            </div>
          </Option>
        </div>
        <div
          className="options"
        >
          <Option
            highlighted={false}
            key="opt-React.js"
            onClick={[Function]}
            selected={true}
            text="React.js"
            tickIcon={true}
            value="React.js"
          >
            <div
              className="selected option"
              key="React.js"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                React.js
              </span>
              <Tick
                className="icoCheck"
              >
                <Sprite
                  className="icoCheck"
                  id="tick"
                  viewBox="0 0 30 30"
                >
                  <svg
                    className="svg-sprite icoCheck"
                    role="img"
                    viewBox="0 0 30 30"
                  >
                    <use
                      xlinkHref="#tick"
                    />
                  </svg>
                </Sprite>
              </Tick>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Vue.js"
            onClick={[Function]}
            selected={false}
            text="Vue.js"
            tickIcon={true}
            value="Vue.js"
          >
            <div
              className="option"
              key="Vue.js"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Vue.js
              </span>
              <Tick
                className="icoCheck"
              >
                <Sprite
                  className="icoCheck"
                  id="tick"
                  viewBox="0 0 30 30"
                >
                  <svg
                    className="svg-sprite icoCheck"
                    role="img"
                    viewBox="0 0 30 30"
                  >
                    <use
                      xlinkHref="#tick"
                    />
                  </svg>
                </Sprite>
              </Tick>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Angular.js"
            onClick={[Function]}
            selected={false}
            text="Angular.js"
            tickIcon={true}
            value="Angular.js"
          >
            <div
              className="option"
              key="Angular.js"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Angular.js
              </span>
              <Tick
                className="icoCheck"
              >
                <Sprite
                  className="icoCheck"
                  id="tick"
                  viewBox="0 0 30 30"
                >
                  <svg
                    className="svg-sprite icoCheck"
                    role="img"
                    viewBox="0 0 30 30"
                  >
                    <use
                      xlinkHref="#tick"
                    />
                  </svg>
                </Sprite>
              </Tick>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-jQuery"
            onClick={[Function]}
            selected={true}
            text="jQuery"
            tickIcon={true}
            value="jQuery"
          >
            <div
              className="selected option"
              key="jQuery"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                jQuery
              </span>
              <Tick
                className="icoCheck"
              >
                <Sprite
                  className="icoCheck"
                  id="tick"
                  viewBox="0 0 30 30"
                >
                  <svg
                    className="svg-sprite icoCheck"
                    role="img"
                    viewBox="0 0 30 30"
                  >
                    <use
                      xlinkHref="#tick"
                    />
                  </svg>
                </Sprite>
              </Tick>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Knockout"
            onClick={[Function]}
            selected={false}
            text="Knockout"
            tickIcon={true}
            value="Knockout"
          >
            <div
              className="option"
              key="Knockout"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Knockout
              </span>
              <Tick
                className="icoCheck"
              >
                <Sprite
                  className="icoCheck"
                  id="tick"
                  viewBox="0 0 30 30"
                >
                  <svg
                    className="svg-sprite icoCheck"
                    role="img"
                    viewBox="0 0 30 30"
                  >
                    <use
                      xlinkHref="#tick"
                    />
                  </svg>
                </Sprite>
              </Tick>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Backbone"
            onClick={[Function]}
            selected={false}
            text="Backbone"
            tickIcon={true}
            value="Backbone"
          >
            <div
              className="option"
              key="Backbone"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Backbone
              </span>
              <Tick
                className="icoCheck"
              >
                <Sprite
                  className="icoCheck"
                  id="tick"
                  viewBox="0 0 30 30"
                >
                  <svg
                    className="svg-sprite icoCheck"
                    role="img"
                    viewBox="0 0 30 30"
                  >
                    <use
                      xlinkHref="#tick"
                    />
                  </svg>
                </Sprite>
              </Tick>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Ember"
            onClick={[Function]}
            selected={false}
            text="Ember"
            tickIcon={true}
            value="Ember"
          >
            <div
              className="option"
              key="Ember"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Ember
              </span>
              <Tick
                className="icoCheck"
              >
                <Sprite
                  className="icoCheck"
                  id="tick"
                  viewBox="0 0 30 30"
                >
                  <svg
                    className="svg-sprite icoCheck"
                    role="img"
                    viewBox="0 0 30 30"
                  >
                    <use
                      xlinkHref="#tick"
                    />
                  </svg>
                </Sprite>
              </Tick>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Bootstrap"
            onClick={[Function]}
            selected={false}
            text="Bootstrap"
            tickIcon={true}
            value="Bootstrap"
          >
            <div
              className="option"
              key="Bootstrap"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Bootstrap
              </span>
              <Tick
                className="icoCheck"
              >
                <Sprite
                  className="icoCheck"
                  id="tick"
                  viewBox="0 0 30 30"
                >
                  <svg
                    className="svg-sprite icoCheck"
                    role="img"
                    viewBox="0 0 30 30"
                  >
                    <use
                      xlinkHref="#tick"
                    />
                  </svg>
                </Sprite>
              </Tick>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Other"
            onClick={[Function]}
            selected={false}
            text="Other"
            tickIcon={true}
            value="Other"
          >
            <div
              className="option"
              key="Other"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Other
              </span>
              <Tick
                className="icoCheck"
              >
                <Sprite
                  className="icoCheck"
                  id="tick"
                  viewBox="0 0 30 30"
                >
                  <svg
                    className="svg-sprite icoCheck"
                    role="img"
                    viewBox="0 0 30 30"
                  >
                    <use
                      xlinkHref="#tick"
                    />
                  </svg>
                </Sprite>
              </Tick>
            </div>
          </Option>
        </div>
        <input
          type="hidden"
          value="React.js,jQuery"
        />
      </div>
    </div>
  </OutsideClickable>
</Select>
`;

exports[`(Component) Select [snapshot] should match 1`] = `
<Select
  multiple={false}
  options={
    Array [
      Object {
        "text": "React.js",
        "value": "React.js",
      },
      Object {
        "text": "Vue.js",
        "value": "Vue.js",
      },
      Object {
        "text": "Angular.js",
        "value": "Angular.js",
      },
      Object {
        "text": "jQuery",
        "value": "jQuery",
      },
      Object {
        "text": "Knockout",
        "value": "Knockout",
      },
      Object {
        "text": "Backbone",
        "value": "Backbone",
      },
      Object {
        "text": "Ember",
        "value": "Ember",
      },
      Object {
        "text": "Bootstrap",
        "value": "Bootstrap",
      },
      Object {
        "text": "Other",
        "value": "Other",
      },
    ]
  }
  required={false}
  selected={
    Array [
      "jQuery",
    ]
  }
  size="medium"
>
  <OutsideClickable
    onOutsideClick={[Function]}
  >
    <div
      className="select"
      key=".0"
      onClick={[Function]}
    >
      <div
        className="selectInner"
        onBlur={[Function]}
        onFocus={[Function]}
        tabIndex={0}
      >
        <div
          className="ph"
        >
          <Option
            closeIcon={false}
            key="jQuery"
            selected={false}
            text="jQuery"
            value="jQuery"
          >
            <div
              className="option"
              key="jQuery"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                jQuery
              </span>
            </div>
          </Option>
        </div>
        <div
          className="options"
        >
          <Option
            highlighted={false}
            key="opt-React.js"
            onClick={[Function]}
            selected={false}
            text="React.js"
            tickIcon={false}
            value="React.js"
          >
            <div
              className="option"
              key="React.js"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                React.js
              </span>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Vue.js"
            onClick={[Function]}
            selected={false}
            text="Vue.js"
            tickIcon={false}
            value="Vue.js"
          >
            <div
              className="option"
              key="Vue.js"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Vue.js
              </span>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Angular.js"
            onClick={[Function]}
            selected={false}
            text="Angular.js"
            tickIcon={false}
            value="Angular.js"
          >
            <div
              className="option"
              key="Angular.js"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Angular.js
              </span>
            </div>
          </Option>
          <Option
            highlighted={true}
            key="opt-jQuery"
            onClick={[Function]}
            selected={true}
            text="jQuery"
            tickIcon={false}
            value="jQuery"
          >
            <div
              className="selected highlighted option"
              key="jQuery"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                jQuery
              </span>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Knockout"
            onClick={[Function]}
            selected={false}
            text="Knockout"
            tickIcon={false}
            value="Knockout"
          >
            <div
              className="option"
              key="Knockout"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Knockout
              </span>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Backbone"
            onClick={[Function]}
            selected={false}
            text="Backbone"
            tickIcon={false}
            value="Backbone"
          >
            <div
              className="option"
              key="Backbone"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Backbone
              </span>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Ember"
            onClick={[Function]}
            selected={false}
            text="Ember"
            tickIcon={false}
            value="Ember"
          >
            <div
              className="option"
              key="Ember"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Ember
              </span>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Bootstrap"
            onClick={[Function]}
            selected={false}
            text="Bootstrap"
            tickIcon={false}
            value="Bootstrap"
          >
            <div
              className="option"
              key="Bootstrap"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Bootstrap
              </span>
            </div>
          </Option>
          <Option
            highlighted={false}
            key="opt-Other"
            onClick={[Function]}
            selected={false}
            text="Other"
            tickIcon={false}
            value="Other"
          >
            <div
              className="option"
              key="Other"
              onClick={[Function]}
            >
              <span
                className="text"
              >
                Other
              </span>
            </div>
          </Option>
        </div>
        <input
          type="hidden"
          value="jQuery"
        />
      </div>
    </div>
  </OutsideClickable>
</Select>
`;
